### ISSUE TITLE
PortFinder crashes with unused adapter

### ISSUE THREAD

**Issue by JasonGoemaat** [angular/angular-cli#2769]
> ## Please provide us with the following information:
### OS?

> Windows 10
### Versions.

> angular-cli: 1.0.0-beta.17
> node: 6.6.0
> os: win32 x64
### Repro steps.

> ng new cli
> cd cli
> npm install
> npm start
### The log given by the failure.

> Error: connect EACCES fe80::dbb0:2b25:9e86:26cd:49152
>    at Object.exports._errnoException (util.js:1036:11)
>    at exports._exceptionWithHostPort (util.js:1059:20)
>    at TCPConnectWrap.afterConnect [as oncomplete](net.js:1080:14)
### Mention any other details that might be useful.

The fe80::xxxxx is a currently disabled or unavailable adapter.  PortFinder.getPort doesn't appear to use the host specified and checks for the port on all adapters, and throws an uncaught exception.  To be able to start my app I had to edit `node_modules/angular-cli/commands/serve.js` and change _checkExpressPort and _ autoFindLiveReloadPort to not use getPort:

```
  return new Promise((resolve, reject) => {
    resolve(commandOptions);
  });
```

> ---
> 
> Thanks! We'll be in touch soon.

**Comment by JasonGoemaat** [comment]
Should have mentioned, it fails when specifying the host and port too, I've tried combinations of ports and different hosts like 127.0.0.1, localhost, and my local network ip:

    ng serve --host 127.0.0.1 --port 4444 --live-reload-port 4445

**Comment by JasonGoemaat** [comment]
I see there is an [issue](https://github.com/ember-cli/ember-cli/issues/6361) on ember-cli for what seems to be the same problem.

**Comment by JasonGoemaat** [comment]
Per a comment, pinning PortFinder at 1.0.7 seems to work around the issue.  Totally understand if someone wants to close this issue, keep the same strategy of using portfinder even if ports are specified, and wait for upstream fixes...

     "portfinder": "1.0.7",

**Comment by eriktrom** [comment]
Portfinder 1.0.9 has been published to npm. It simply reverts 1.0.8 back to 1.0.7, so the issue is fixed now.

Sorry about that folks

**Comment by filipesilva** [comment]
Heya @eriktrom, thanks for the quick fix! I added `1.0.9` explicitely in https://github.com/angular/angular-cli/pull/2831 now.
